amp.php

```diff
@@ -157,3 +160,20 @@ function _amp_bootstrap_customizer() {
 	}
 }
 add_action( 'plugins_loaded', '_amp_bootstrap_customizer', 9 );
+
+/**
+ * Redirects the old AMP URL to the new AMP URL.
+ * If post slug is updated the amp page with old post slug will be redirected to the updated url.
+ *
+ * @param  string $link New URL of the post.
+ *
+ * @return string $link URL to be redirected.
+ */
+function amp_redirect_to_new_url( $link ) {
+
+	if ( is_amp_endpoint() ) {
+		$link = trailingslashit( $link ) . AMP_QUERY_VAR;
```

Comment:
You are right that using `AMP_QUERY_VAR` does protect us in case someone changes the value from `amp` to something else. However, the plugin does provide a filter that let's you change the AMP url so it could look very different from the default, e.g. `/amp/2017/01/01/this-post-is-great/`

This is a bit of an edge case though, so I think we can merge as-is for now.